### ISSUE TITLE
tvshow, episode, artist and album.nfo's scanned into Movie Library

### ISSUE THREAD

**Issue by KarellenX** [xbmc/xbmc#16112]
### Describe the bug
Kodi is not checking xml tags to ensure the found nfo is correct for the current library (movie, tv show, episode, music videos)

- I can place the **tvshow.nfo** into a movie folder, and scan it into my movie library.
- I can place the **American Gods S01E01.nfo** episode nfo into my movie library and scan it into the movie library.
- I can also scan movie.nfo files into the TV Show library. Though this does require changing the filename to tvshow.nfo. 

Scanning a movie.nfo into an episode does not work as expected.

In all cases this indicates that the parent tags are being ignored...
`<movie></movie>  <tvshow></tvshow>  <episode></episode>`

It also indicated that the nfo file name does not need to match the video filename, which seems to be another problem.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
1. The scan should fail if the wrong class of NFO file is used or 
2. The scrapers should scrape online information after ignoring the NFO File

depending on whether a scraper or Local Information Only is set.

## Actual Behavior
Kodi will scan the tvshow.nfo and the episode.nfo into the movie library

Kodi will scan the movie.nfo (renamed to tvshow.nfo) into the TV Show library

### To Reproduce

1. Locate/create Movie Folder that is not yet scraped into the movie library
2. Place the **tvshow.nfo** or the episode.nfo into a movie folder, 
3. Scan the movie into the library using your preferred method

**NOTE:**  The nfo file name does not need to match the video filename, which seems to be another problem.

### Debuglog
Log is here... https://paste.kodi.tv/uwokapakor.kodi

Pertinent Extracts from the log

> _DEBUG: VideoInfoScanner: Found matching full NFO file: C:\KODI\Test- Movies\Ant-Man (2015)\American Gods S01E01.nfo_

Line 926

> _DEBUG: VideoInfoScanner: Found matching full NFO file: C:\KODI\Test- Movies\Ant-Man (2015)\tvshow.nfo

Line 1154

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 10 1803
 - Kodi version: 18.3-RC1 Git:20190504-a7d4df4d5b

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by KarellenX** [comment]
Further testing reveals that **artist.nfo** and **album.nfo** are scanned into the Movie Library. I did not test on the TV Show library, but expect it would scan there also.

Movie nfo files do not scan into the Music Library

![MusicInMovies](https://user-images.githubusercontent.com/44626605/57589165-5d5b7e00-7563-11e9-805f-2cfdd2bfb0c8.jpg)

**Comment by enen92** [comment]
@KarellenX can you annex .nfo examples as well?

**Comment by KarellenX** [comment]
Hello @enen92 

All nfo files are created by Kodi via export. Initial scrape was done by Kodi scrapers.

I have uploaded as follows for movies. If you also want TV Shows, let me know and I can upload also, but same result...

Zip file with movie folder and files. Extract and scan (9 kb) ... https://ufile.io/orqq57dz

If you prefer just the nfo files...
artist.nfo ... https://paste.kodi.tv/mipicoyamo
album.nfo ... https://paste.kodi.tv/dohimosugo
American Gods S01E01.nfo ... https://paste.kodi.tv/niguhagupa
tvshow.nfo ... https://paste.kodi.tv/lafofefazi

Once you scan, you should end up with this
Left to right... Artist, Episode, Album and TV Show

![nfo](https://user-images.githubusercontent.com/44626605/58057389-03cd0200-7ba8-11e9-82ba-08185ca2a3c3.jpg)

**Comment by KarellenX** [comment]
@rmrector Is the variation in the artwork sizing normal? I vaguely recall you submitted a PR to ensure only Posters were used in displays for Movies, as landscape images were sometimes being used.

**Comment by rmrector** [comment]
Yeah. If your NFO files have "thumb" images specified they will be assigned to these "movies", and without a poster image Estuary will fall back to that thumb image. Looks goofy but it's doing the best it can with these odd NFOs you're feeding it 😄.

That PR changed it to prefer posters, but it can still fall back to "thumb".

**Comment by KarellenX** [comment]
Goofy... haha. All kodi created, but yeah, nfo files being read into the wrong libraries.

Thanks, understood the fallback.

**Comment by KarellenX** [comment]
Have also discovered that removing the `<movie></movie>` and `<tvshow></tvshow>` tags from the respective nfo files will create an empty record in the database when scanned.

**Comment by monisriz** [comment]
Could this bug be used to add two sources with different content types - one containing "Movies" and other "Tv shows" to the same Library view? I tried recreating the downloading the folder structure @KarellenX linked to but it seems the link is old. I am able to mix "movie" and "episode" but not movie and tvshow.

Even though this is a bug, it got me thinking of the problem I face when dealing with Documentaries - Episodic (e.g. Dynasties) and Feature length (e.g. Apollo 11). I sort of get around it by using a playlist for the feature length docs and including a virtual folder which points to the folder containing Episodic docs. But it's a bit ugly as the virtual folder doesn't have any artwork/image and all the docs inside it are hidden until that folder is opened.

**Comment by KarellenX** [comment]
>Could this bug be used to add two sources with different content types - one containing "Movies" and other "Tv shows" to the same Library view? 

Not sure of the actual process you are trying to use here, but if you simply move a movie and nfo file into a tv show as an episode and rename it with SxxExx, then it will scan.


>I am able to mix "movie" and "episode" but not movie and tvshow.

A tv show is a collection of episodes. There is no video file for "tvshow", it is just a grouping, so how are you trying to change a movie into a tv show?

**Comment by monisriz** [comment]
> > Could this bug be used to add two sources with different content types - one containing "Movies" and other "Tv shows" to the same Library view?
> 
> Not sure of the actual process you are trying to use here, but if you simply move a movie and nfo file into a tv show as an episode and rename it with SxxExx, then it will scan.
> 
> > I am able to mix "movie" and "episode" but not movie and tvshow.
> 
> A tv show is a collection of episodes. There is no video file for "tvshow", it is just a grouping, so how are you trying to change a movie into a tv show?

Ok I clearly was not good with my explanation earlier.
So the reason why this issue is of interest to me is that I have been trying to have a single menu entry for "Documentaries" under which there could be:
1. Documentary films (e.g. Apollo 11 or Blackfish) with their "movie" nfo.
2. Documentary series (e.g. Dynasties) displayed with a single entry point (like it is for tv shows - Westworld in your screenshot above) under which it is broken down into seasons and episodes.

Now the issue is that when creating a source Kodi wants us to select Content type so doc films and doc series can't be combined in one source. That's fine - split the sources and create two separate sources (tv shows and movies). In order to combine these two sources, I have tried creating a playlist and adding both the sources but again Kodi asks to choose content type as tv show or movie and choosing either one of these types will not pick up the other type. If this issue can't fix this, do you have any suggestion or a way to combine two content types into a single playlist?

From the issue description I thought perhaps this issue will some how fix this issue for me (and maybe other users). Hope I was better with clarification this time around.

**Comment by KarellenX** [comment]
Ok, I understand now.

There is no way to combine TV Shows and Movies into a single node or smart playlist. This bug will not help you achieve that.

In a situation like yours I just move the movie into the TV Show. For example single feature movies that are prequels or sequels to TV shows, such as Stargate (1994) or Serenity (2005), I move them into the TV show as the first or final episode of the show. Rename the movie file to include SxxExx and adjust the nfo file by changing `<movie></movie>` to `<episode></episode>`. Most of the tags are the same and any different tags are ignored. 

This bug is not reading the parent (eg `<movie></movie>`) tags and ignoring the filename, so you no longer even need to change the parent tags.

Or create a new TV Show to collate all the movies in one location. Just include the SxxExx in the filename and that should work, without changing anything in the nfo file.

The only other thing I can think of that may work, but I have not tried it, is creating a Basic Playlist using something like this add-on... 
https://forum.kodi.tv/showthread.php?tid=13752&pid=2216160#pid2216160

or maybe this...
https://kodi.tv/addon/music-add-ons-picture-add-ons-plugins-video-add-ons/simpleplaylists

**Comment by KarellenX** [comment]
Users are starting to notice this bug and, annoyingly, are using it as an excuse for their defective Media Manager created nfo files. 

https://forum.kodi.tv/showthread.php?tid=348648&pid=2894857#pid2894857
and
https://forum.kodi.tv/showthread.php?tid=348590&pid=2894144#pid2894144
